presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/OptimizeMixedDistinctAggregations.java

```diff
@@ -210,6 +210,7 @@ public PlanNode visitMarkDistinct(MarkDistinctNode node, RewriteContext<Optional
             allSymbols.addAll(groupBySymbols);
             allSymbols.addAll(nonDistinctAggregateSymbols);
             allSymbols.add(distinctSymbol);
+            allSymbols = allSymbols.stream().distinct().collect(Collectors.toList());
```

Comment:
@martint  Interestingly , after rebase, only deduplicating `groupByKeys` is required  -- if groupBy Column = Distinct Column, the distinct column will be automatically renamed to `col_2` . I assume some upstream changes in Planner did this rename.

I still converting `groupByKeys` into a `Set` since I think we shouldn't rely on the renaming behavior  in previous optimizer / planner.